framework/db/Query.php

```diff
@@ -466,18 +466,29 @@ public function getTablesUsedInFrom()
     {
         if (empty($this->from)) {
             return [];
-        } elseif (is_array($this->from)) {
+        }
+
+        if (is_array($this->from)) {
             $tableNames = $this->from;
         } elseif (is_string($this->from)) {
             $tableNames = preg_split('/\s*,\s*/', trim($this->from), -1, PREG_SPLIT_NO_EMPTY);
+        } elseif ($this->from instanceof Expression) {
+            $tableNames = [$this->from];
         } else {
             throw new InvalidConfigException(gettype($this->from) . ' in $from is not supported.');
         }
 
         // Clean up table names and aliases
         $cleanedUpTableNames = [];
         foreach ($tableNames as $alias => $tableName) {
-            if (!is_string($alias)) {
+            $tableNameString = null;
+            if (is_string($tableName)) {
+                $tableNameString = $tableName;
+            } elseif ($tableName instanceof Expression) {
+                $tableNameString = $tableName->expression;
```

Comment:
Again, in terms of that PR, this code support only `Expression` class, or any `ExpressionInterface`?

Reply:
`ExpressionInterface` isn't in master yet.

Reply:
I think in this case it's about class.